Windows tests started failing on CI today across all Python versions

## Description

```
FAILED tests/auth/test_authorizer.py::test_authorized_requests[True-POST-/api/terminals-] - AttributeError: type object 'object' has no attribute 'spawn'
FAILED tests/auth/test_authorizer.py::test_authorized_requests[True-GET-/api/terminals-None] - AttributeError: type object 'object' has no attribute 'spawn'
FAILED tests/auth/test_authorizer.py::test_authorized_requests[True-GET-/terminals/websocket/{term_name}-None] - AttributeError: type object 'object' has no attribute 'spawn'
FAILED tests/auth/test_authorizer.py::test_authorized_requests[True-DELETE-/api/terminals/{term_name}-None] - AttributeError: type object 'object' has no attribute 'spawn'
FAILED tests/auth/test_authorizer.py::test_authorized_requests[False-POST-/api/terminals-] - AttributeError: type object 'object' has no attribute 'spawn'
FAILED tests/auth/test_authorizer.py::test_authorized_requests[False-GET-/api/terminals-None] - AttributeError: type object 'object' has no attribute 'spawn'
FAILED tests/auth/test_authorizer.py::test_authorized_requests[False-GET-/terminals/websocket/{term_name}-None] - AttributeError: type object 'object' has no attribute 'spawn'
FAILED tests/auth/test_authorizer.py::test_authorized_requests[False-DELETE-/api/terminals/{term_name}-None] - AttributeError: type object 'object' has no attribute 'spawn'
FAILED tests/test_terminal.py::test_terminal_create - tornado.httpclient.HTTPClientError: HTTP 500: Internal Server Error
FAILED tests/test_terminal.py::test_terminal_create_with_kwargs - tornado.httpclient.HTTPClientError: HTTP 500: Internal Server Error
FAILED tests/test_terminal.py::test_terminal_create_with_cwd - tornado.httpclient.HTTPClientError: HTTP 500: Internal Server Error
FAILED tests/test_terminal.py::test_culling - tornado.httpclient.HTTPClientError: HTTP 500: Internal Server Error
```

<img width="1482" height="505" alt="Image" src="https://github.com/user-attachments/assets/8d4a3e04-8c92-4208-87f9-4a0edbc4cc83" />

https://github.com/jupyter-server/jupyter_server/actions/runs/27266074745/job/80523334363

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -33,7 +33,11 @@ dependencies = [
     "nbformat>=5.3.0",
     "packaging>=22.0",
     "prometheus_client>=0.9",
-    "pywinpty>=2.0.1;os_name=='nt'",
+    # pywinpty 3.0.4 ships broken Windows wheels: winpty.dll (amd64) and all native
+    # binaries (arm64) are no longer bundled, so the .pyd fails to load -> import winpty
+    # raises ImportError -> terminado falls back to `object` -> object.spawn AttributeError.
+    # See https://github.com/andfoy/pywinpty/issues/573 (regressed in 3.0.4, 2026-06-10).
+    "pywinpty>=2.0.1,!=3.0.4;os_name=='nt'",
     "pyzmq>=24",
     "Send2Trash>=1.8.2",
     "terminado>=0.8.3",

